api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -274,28 +275,58 @@ func (mw *azureManagedControlPlaneWebhook) ValidateDelete(ctx context.Context, o
 
 // Validate the Azure Managed Control Plane and return an aggregate error.
 func (m *AzureManagedControlPlane) Validate(cli client.Client) error {
+	var allErrs field.ErrorList
 	validators := []func(client client.Client) error{
-		m.validateName,
-		m.validateVersion,
 		m.validateSSHKey,
-		m.validateLoadBalancerProfile,
 		m.validateAPIServerAccessProfile,
-		m.validateManagedClusterNetwork,
-		m.validateAutoScalerProfile,
 		m.validateIdentity,
 		m.validateNetworkPluginMode,
 		m.validateDNSPrefix,
 		m.validateDisableLocalAccounts,
 	}
-
-	var errs []error
 	for _, validator := range validators {
 		if err := validator(cli); err != nil {
-			errs = append(errs, err)
+			allErrs = append(allErrs, field.InternalError(field.NewPath("spec"), err))
```

Comment:
Ah good catch. It would definitely cause an unecessary wrap around the existing errors.

Reply:
Since now this is a `field.ErrorList`, I can't just append an error of type `error` to `allErrs`, so I'll be refactoring all of the validators to return a `field.ErrorList` instead.